File: code_examples/3-Structured_Problem_Solving/Scope of Variables/scopeOfVariables.swift
// Copyright (c) Jonathan Pasco-Arnone All rights reserved.
//
// Created by: Jonathan Pasco-Arnone
// Created on: Mar 2022
// This program shows how local and global variables work

// Global variable
let variableX = 25

// This shows what happens with local variables
func localVariable() {
    let variableX = 10
    let variableY = 30
    let variableZ = variableX + variableY
    print("Local variableX, variableY, variableZ: ",
        variableX, " + ", variableY, " = ", variableZ)
}
// This shows what happens with local variables
func globalVariable() {
    let variableX = 10
    let variableY = 30
    let variableZ = variableX + variableY
    print("Global variableX. Local variableY, variableZ: ",
        variableX, " + ", variableY, " = ", variableZ)
}

localVariable()
globalVariable()

File: code_examples/3-Structured_Problem_Solving/Scope of Variables/scopeOfVariables.java
// Copyright (c) Jonathan Pasco-Arnone All rights reserved.
//
// Created by: Jonathan Pasco-Arnone
// Created on: Mar 2022
// This program shows how local and global variables work

class Main {

    /**
    * Global variable
    */
    private static int variableX = 25;

    /**
    * This shows what happens with local variables
    */
    public static void localVariable() {
        int variableX = 10;
        int variableY = 30;
        int variableZ = variableX + variableY;
        System.out.println("Local variableX, variableY, variableZ: "
            + variableX + " + " + variableY + " = " + variableZ);
        
    }

    /**
    * This shows what happens with global variables    */
    public static void globalVariable() {
        variableX = variableX + 1;
        int variableY = 30;
        int variableZ = variableX + variableY;
        System.out.println("Global variableX. Local variableY, variableZ: "
            + variableX + " + " + variableY + " = " + variableZ);
        
    }

    /**
    * This function calls local and global
    */
    public static void main(String[] args) {
        localVariable();
        globalVariable();
    }
}
